capabilities/writetarget/write_target.go

```diff
@@ -0,0 +1,510 @@
+	info.reportInfo.reportID = binary.BigEndian.Uint16(inputs.ID)
+
+	err = c.beholder.ProtoEmitter.EmitWithLog(ctx, builder.buildWriteInitiated(info))
+	if err != nil {
+		c.lggr.Errorw("failed to emit write initiated", "err", err)
+	}
+
+	// Check whether the report is valid (e.g., not empty)
+	if len(inputs.Report) == 0 {
+		// We received any empty report -- this means we should skip transmission.
+		err = c.beholder.ProtoEmitter.EmitWithLog(ctx, builder.buildWriteSkipped(info, "empty report"))
+		if err != nil {
+			c.lggr.Errorw("failed to emit write skipped", "err", err)
+		}
+		return success(), nil
+	}
+
+	// Update the info with the report info
+	info.reportInfo = &reportInfo{
+		reportID:      info.reportInfo.reportID,
+		reportContext: inputs.Context,
+		report:        inputs.Report,
+		signersNum:    uint32(len(inputs.Signatures)),
+	}
+
+	// Decode the report
+	reportDecoded, err := platform.Decode(inputs.Report)
+	if err != nil {
+		msg := builder.buildWriteError(info, 0, "failed to decode the report", err.Error())
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	}
+
+	// Validate encoded report is prefixed with workflowID and executionID that match the request meta
+	if reportDecoded.ExecutionID != request.Metadata.WorkflowExecutionID {
+		msg := builder.buildWriteError(info, 0, "decoded report execution ID does not match the request", "")
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	} else if reportDecoded.WorkflowID != request.Metadata.WorkflowID {
+		msg := builder.buildWriteError(info, 0, "decoded report workflow ID does not match the request", "")
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	}
+
+	// Fetch the latest head from the chain (timestamp), retry with a default backoff strategy
+	ctx = retry.CtxWithID(ctx, info.request.Metadata.WorkflowExecutionID)
+	head, err := retry.With(ctx, c.lggr, c.cs.LatestHead)
+	if err != nil {
+		msg := builder.buildWriteError(info, 0, "failed to fetch the latest head", err.Error())
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	}
+
+	c.lggr.Debugw("non-empty valid report",
+		"reportID", info.reportInfo.reportID,
+		"report", "0x"+hex.EncodeToString(inputs.Report),
+		"reportLen", len(inputs.Report),
+		"reportDecoded", reportDecoded,
+		"reportContext", "0x"+hex.EncodeToString(inputs.Context),
+		"reportContextLen", len(inputs.Context),
+		"signaturesLen", len(inputs.Signatures),
+		"executionID", request.Metadata.WorkflowExecutionID,
+	)
+
+	state, err := c.targetStrategy.QueryTransmissionState(ctx, info.reportInfo.reportID, request)
+
+	if err != nil {
+		msg := builder.buildWriteError(info, 0, "failed to fetch [TransmissionState]", err.Error())
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	}
+
+	switch state.Status {
+	case TransmissionStateNotAttempted:
+		c.lggr.Debugw("Transmission not attempted yet, retrying", "reportID", info.reportInfo.reportID)
```

Comment:
```suggestion
		c.lggr.Debugw("Report is not on chain yet, pushing tx", "reportID", info.reportInfo.reportID)
```
`Retrying` is not accurate, as it might be  our first attempt